### DISCUSSION TITLE
How to change environment variable of an existing container

### DISCUSSION THREAD

**Discussion by 1player** [containers/podman#15044] [Q&A]
I have a container which I would really like not to recreate with a custom `--env` option.

Is there a way for me to change a container's environment option without having to recreate it?

**Comment by rhatdan** [comment]
No, you have to clone it.

**Reply by 1player** [reply]
There doesn't seem to be an option to change the env variables with `podman container clone`, what am I missing?

**Reply by rhatdan** [reply]
@cdoern PTAL

**Reply by mheon** [reply]
This is an interesting one because `container clone` so far has only been additive, it doesn't really allow subtracting pieces.

Seems like a good potential usecase for `podman generate spec`?

**Reply by cdoern** [reply]
> This is an interesting one because `container clone` so far has only been additive, it doesn't really allow subtracting pieces.
> 
> Seems like a good potential usecase for `podman generate spec`?

technically you can "replace" stuff with container clone, like you can do `--cpus=4` instead of `--cpus-5` so this could just be a more extreme version of replacing a config entity. 

I could see this being applicable to both clone and generate spec

**Reply by rhatdan** [reply]
We do have --unsetenv?

**Reply by rhatdan** [reply]
Although this does not work the way I was hoping.
 $ podman run --env=apple=orange --unsetenv apple fedora printenv apple
orange

**Reply by 1player** [reply]
Honestly my issue is that this is a *pet* container, i.e. I don't want to lose any data, if possible. I know how it's been created (I can get all the details with a `podman generate systemd --new`), so if there's a way of exporting a tarball of its contents, creating a new one with a different set of --env options and reimporting the tarball, I can try that.

Can `podman export` help?

**Reply by rhatdan** [reply]
You could commit the container to an image and then create a new container off of the image.
